src/Raven.Server/Documents/Indexes/Workers/HandleReferences.cs

```diff
@@ -134,33 +131,22 @@ private unsafe bool HandleDocuments(ActionType actionType, DocumentsOperationCon
 
                                         references = _documentsStorage
                                             .GetDocumentsFrom(databaseContext, referencedCollection.Name, lastEtag + 1, 0, pageSize)
-                                            .Select(document =>
-                                            {
-                                                _reference.Key = document.Id;
-                                                _reference.Etag = document.Etag;
-
-                                                return _reference;
-                                            });
+                                            .Select(document => new Reference{Etag = document.Etag, Key = document.Id });
                                         break;
                                     case ActionType.Tombstone:
                                         if (lastCollectionEtag == -1)
                                             lastCollectionEtag = _index.GetLastTombstoneEtagInCollection(databaseContext, collection);
 
                                         references = _documentsStorage
                                             .GetTombstonesFrom(databaseContext, referencedCollection.Name, lastEtag + 1, 0, pageSize)
-                                            .Select(tombstone =>
-                                            {
-                                                _reference.Key = tombstone.LowerId;
-                                                _reference.Etag = tombstone.Etag;
-
-                                                return _reference;
-                                            });
+                                            .Select(tombstone => new Reference { Etag = tombstone.Etag, Key = tombstone.LowerId });
                                         break;
                                     default:
                                         throw new NotSupportedException();
                                 }
 
-                                foreach (var referencedDocument in references)
+                                var list = references.ToList();
+                                foreach (var referencedDocument in list)
```

Comment:
Why this is needed? Looks like a debugging code?

Reply:
done